### PR TITLE
chore(TDI-38415): Add documentation for NetSuite components

### PR DESCRIPTION
**Pull request by iistomin-gl**
**What is the current behavior?** (You can also link to an open issue here)
- NetSuite related classes don't have proper documentation

**What is the new behavior?**
- NetSuite related classes have documentation

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [X] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [X] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [X] Other... Please describe: Documentation

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

**Other information**:
JIRA issue: https://jira.talendforge.org/browse/TDI-38415

### CODE REVIEW

File: components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/output/NsObjectOutputTransducer.java [hunk 1]
```diff
@@ -66,58 +75,30 @@ public void setReference(boolean reference) {
...
         if (reference) {
+            // If target NetSuite data object is record ref then
+            // we should get descriptor for RecordRef type.
             typeDesc = metaDataSource.getTypeInfo(recordTypeInfo.getRefType().getTypeName());
         } else {
             typeDesc = metaDataSource.getTypeInfo(typeName);
         }
     }
 
-    public NsRef getRef(IndexedRecord indexedRecord) {
```
Comment (iistomin-gl) [inline components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/output/NsObjectOutputTransducer.java]: This code is not used and was removed.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/414/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/414/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/433/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/433/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/435/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/435/)

**Pull request merged**